### PR TITLE
Add config migrate for old Mind memory setup

### PR DESCRIPTION
**Pull request by basnijholt**
## Summary
- Add `mindroom config migrate` for safe text-preserving config migrations.
- Migrate the exact old `config init` Mind memory wiring from `mind_memory` knowledge base search to the new `memory` tool plus semantic `memory.search` block.
- Leave customized `mind_memory` paths untouched instead of guessing.

## Tests
- `uv run pytest tests/test_cli_config.py::TestConfigMigrate -q`
- `uv run pytest tests/test_cli_config.py -q`
- `uv run ruff check src/mindroom/cli/config.py tests/test_cli_config.py`
- `uv run ruff format --check src/mindroom/cli/config.py tests/test_cli_config.py`
- `uv run ty check src/mindroom/cli/config.py tests/test_cli_config.py`
- `uv run pre-commit run --all-files`

Full suite note: `uv run pytest` currently fails only `tests/test_live_message_coalescing.py::test_image_and_text_coalesce_into_single_dispatch`; rerunning that test alone reproduces the same failure. This PR only touches config CLI files and the vulture whitelist.

## Summary by CodeRabbit

* **New Features**
  * Added `mindroom config migrate` subcommand to automatically migrate existing configurations from legacy memory integration to the newer semantic memory search structure.

* **Documentation**
  * Updated CLI reference documentation to include the new migrate subcommand.

* **Tests**
  * Added comprehensive test coverage for config migration scenarios, including validation of successful migrations and edge cases.

### CODE REVIEW

File: src/mindroom/cli/config.py [hunk 1]
```diff
@@ -467,6 +562,39 @@ def config_init(
...
+    try:
+        content = config_file.read_text(encoding="utf-8")
+    except (OSError, UnicodeError) as exc:
+        format_validation_errors(exc, config_path=config_file)
+        raise typer.Exit(1) from None
+
+    migrated, migrated_mind_memory = _migrate_old_config_init_mind_memory(content)
+    if not migrated_mind_memory:
+        console.print("[green]No migrations applied.[/green]")
+        return
```
[the remark this answers is not in the document]
Comment (basnijholt) [reply src/mindroom/cli/config.py]: Fixed in 2db012c74: migration writes now use a sibling temp file, fsync the full content, preserve the existing file mode, then atomically replace the config.

File: tests/test_cli_config.py [hunk 2]
```diff
@@ -1036,6 +1036,228 @@ def test_init_vertexai_claude_preset_uses_vertex_models(self, tmp_path: Path) ->
...
+        )
+        .replace(
+            """\
+  file:
+    max_entrypoint_lines: 200
+  auto_flush:
+""",
+            """\
+  file:
+    max_entrypoint_lines: 200
```
[the remark this answers is not in the document]
Comment (basnijholt) [reply tests/test_cli_config.py]: Fixed in 2db012c74: expected migrated config is now an explicit literal instead of mirroring production replacements. Added write-failure regression coverage too.

File: src/mindroom/cli/config.py [hunk 3]
```diff
@@ -467,6 +562,39 @@ def config_init(
...
+
+    try:
+        config_file.write_text(migrated, encoding="utf-8")
+    except OSError as exc:
+        format_validation_errors(exc, config_path=config_file)
+        raise typer.Exit(1) from None
```
[the remark this answers is not in the document]
Comment (basnijholt) [reply src/mindroom/cli/config.py]: Fixed in 2db012c74: write failures now use a write-specific CLI error instead of config validation formatting. Migrated config writes now go through the atomic writer too.
